pkg/k8s/portforward.go

```diff
@@ -162,6 +164,42 @@ func (pf *PortForward) Run() error {
 	return fw.ForwardPorts()
 }
 
+// Init creates and runs a port-forward connection.
+// This function blocks until the connection is established, in which case it returns nil.
+// It's the caller's responsibility to call Stop() to finish the connection.
+func (pf *PortForward) Init() error {
+	log.Debugf("Starting port forward to %s %d:%d", pf.url, pf.localPort, pf.remotePort)
+
+	failure := make(chan error)
+
+	go func() {
+		if err := pf.run(); err != nil {
+			failure <- err
+		}
+
+		select {
+		case <-pf.GetStop():
+			// stopCh was closed, do nothing
+		default:
+			// pf.run() returned for some other reason, close stopCh
```

Comment:
Is this expected to ever happen?  Should we log/warn here?

Reply:
This can happen if the connection unexpectedly closes:
https://github.com/kubernetes/client-go/blob/2c6e35a5b9cf292580c2d8ecb7d141572997d3cd/tools/portforward/portforward.go#L232
... though I wouldn't be opposed to logging here.

Reply:
Yeah I confirmed this is already being logged by k8s:
```bash
$ bin/go-run cli dashboard
Linkerd dashboard available at:
http://127.0.0.1:50750
Grafana dashboard available at:
http://127.0.0.1:50750/grafana
Opening Linkerd dashboard in the default browser
[27967:27986:0626/101432.351720:ERROR:browser_process_sub_thread.cc(221)] Waited 3 ms for network service
Opening in existing browser session.

# I'm killing Minkube here...

E0626 10:14:37.566713   27828 portforward.go:224] lost connection to pod
E0626 10:14:37.566770   27828 portforward.go:224] lost connection to pod
```